gym_xiangqi/test/piece_test.py

```diff
@@ -21,6 +35,84 @@ def test_piece_initialization(self):
             piece.color = 1
             self.assertEqual(piece.color, BLACK)
 
+    def test_general_can_move_within_palace(self):
+        env = XiangQiEnv()
+        # Test red general can only move forward 1 position from
+        # starting position.
+        self.diff_move_list(
+            env=env,
+            piece_id=GENERAL,
+            expected=[(GENERAL, [9, 4], [8, 4])]
+        )
+
+        # Move general to the center of the palace.
```

Comment:
@hojoung97, the way I am moving the pieces here also is temporary, feel free to remove them once you have the `step` method.

Reply:
sounds good 👍